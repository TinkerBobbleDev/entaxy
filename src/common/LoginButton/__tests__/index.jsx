import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import { LoginButtonComponent } from '../'

describe('HandleLogin', () => {
  const mochHandleLogin = jest.fn()
  const mochHandleLogout = jest.fn()

  beforeEach(() => {
    jest.clearAllMocks()
  })

  it('matches snapshot with logged out user', () => {
    const component = renderer.create((
      <LoginButtonComponent
        user={{ isAuthenticated: false }}
        handleLogin={mochHandleLogin}
        handleLogout={mochHandleLogout}
      />
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('matches snapshot with logged in user profile', () => {
    const component = renderer.create((
      <LoginButtonComponent
        user={{ isAuthenticated: true, name: 'Test name' }}
        handleLogin={mochHandleLogin}
        handleLogout={mochHandleLogout}
      />
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('handles login', () => {
    const wrapper = mount((
      <LoginButtonComponent
        user={{ isAuthenticated: false }}
        handleLogin={mochHandleLogin}
        handleLogout={mochHandleLogout}
      />
    ))
    expect(wrapper.find('button > span').text()).toBe('Login')
    wrapper.find('button').simulate('click')
    expect(mochHandleLogin).toHaveBeenCalled()
    expect(mochHandleLogout).not.toHaveBeenCalled()
  })


  it('handles logout', () => {
    const wrapper = mount((
      <LoginButtonComponent
        user={{ isAuthenticated: true, name: 'Test user' }}
        handleLogin={mochHandleLogin}
        handleLogout={mochHandleLogout}
      />
    ))
    expect(wrapper.find('button > span').text()).toBe('Test user Logout')
    wrapper.find('button').simulate('click')
    expect(mochHandleLogin).not.toHaveBeenCalled()
    expect(mochHandleLogout).toHaveBeenCalled()
  })
})
