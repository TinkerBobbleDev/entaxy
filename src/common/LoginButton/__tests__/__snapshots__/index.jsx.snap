// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HandleLogin matches snapshot with logged in user profile 1`] = `
Array [
  <div
    className={undefined}
  >
    <div
      className="MuiAvatar-root-27"
    />
    <button
      aria-owns={null}
      className="MuiButtonBase-root-17 MuiButton-root-1 MuiButton-colorInherit-6"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-2"
      >
        Test name
      </span>
      <span
        className="MuiTouchRipple-root-20"
      />
    </button>
  </div>,
  <div
    data-placement={undefined}
    style={
      Object {
        "left": 0,
        "opacity": 0,
        "pointerEvents": "none",
        "position": "absolute",
        "top": 0,
        "zIndex": 1,
      }
    }
  >
    <div
      className="MuiPaper-root-30 MuiPaper-elevation2-34 MuiPaper-rounded-31"
      id="menu-list-grow"
      style={
        Object {
          "opacity": 0,
          "transform": "scale(0.75, 0.5625)",
          "transformOrigin": "0 0 0",
        }
      }
    >
      <ul
        className="MuiList-root-57 MuiList-padding-58"
        onBlur={[Function]}
        onKeyDown={[Function]}
        role="menu"
      >
        <li
          className="MuiButtonBase-root-17 MuiListItem-root-63 MuiListItem-default-66 MuiListItem-gutters-70 MuiListItem-button-71 MuiMenuItem-root-61"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="menuitem"
          tabIndex={0}
        >
          Profile
          <span
            className="MuiTouchRipple-root-20"
          />
        </li>
        <li
          className="MuiButtonBase-root-17 MuiListItem-root-63 MuiListItem-default-66 MuiListItem-gutters-70 MuiListItem-button-71 MuiMenuItem-root-61"
          onBlur={[Function]}
          onClick={[MockFunction]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="menuitem"
          tabIndex={-1}
        >
          Logout
          <span
            className="MuiTouchRipple-root-20"
          />
        </li>
      </ul>
    </div>
  </div>,
]
`;

exports[`HandleLogin matches snapshot with logged out user 1`] = `
<button
  className="MuiButtonBase-root-25 MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6 MuiButton-colorInherit-20"
  disabled={false}
  onBlur={[Function]}
  onClick={[MockFunction]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button"
>
  <span
    className="MuiButton-label-2"
  >
    Login
  </span>
  <span
    className="MuiTouchRipple-root-28"
  />
</button>
`;
